sources/packages/api/src/services/restriction/student-restriction.service.ts

```diff
@@ -268,4 +223,40 @@ export class StudentRestrictionService extends RecordDataModelService<StudentRes
     } as Note;
     return this.repo.save(studentRestrictionEntity);
   }
+  /**
+   * The service function checks if the requested student has
+   * the restricted restriction.
+   * @param studentId student Id
+   * @param restrictionActions list of restriction actions
+   * @param checkAll,is by default false, checkAll decides if the all the
+   * elements in the restrictionActions should be checked (i.e checkAll = true)
+   * or any one of the element is to be checked (i.e checkAll = false).
+   * @returns boolean, true if the restriction action is present
+   * for the student, else false.
+   */
+  async isRestrictionActionExists(
```

Comment:
I would recommend naming it as `hasRestrictionAction`. It sounds clear IMO but it is just a suggestion 😉